### PR TITLE
Fail the CI build if there are any uncommitted changes

### PR DESCRIPTION
**Pull request by jeffkl**
I'll only be updating cibuild.cmd so ignore the other change as I'm using it to test this.

### CODE REVIEW

File: cibuild.cmd [hunk 1]
```diff
@@ -156,6 +156,20 @@ if %ERRORLEVEL% NEQ 0 (
     goto :error
 )
 
+:: Only detect source control changes when running in the CI environment
+:: Detect if there are any changed files which should fail the build
+if DEFINED JENKINS_URL (
+    echo Detecting changed files...
+    git diff --cached HEAD --exit-code
```
Comment (rainersigwald) [inline cibuild.cmd]: Isn't a failure here ignored since you go to the next line?

Comment (rainersigwald) [inline cibuild.cmd]: You could maybe do `git diff --exit-code HEAD` which should show all diffs between the work tree and HEAD.

Torn on `--quiet` versus `--exit-code`--it'd be nice to get a list on failure, so I guess I prefer `--exit-code`.

Comment (jeffkl) [inline cibuild.cmd]: I'm no git expert, but it worked when I tested it.  I'm going to try this instead:

`git --no-pager diff HEAD --word-diff=plain --exit-code` which seems to work...

**Comment by cdmihai** [comment]
Also print out the diff for ease of understanding :)

**Comment by jeffkl** [comment]
@dotnet-bot test Windows_NT for Desktop please

**Comment by jeffkl** [comment]
@dotnet-bot test this please

**Pull request merged**